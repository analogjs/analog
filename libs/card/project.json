{
  "name": "card",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/card/src",
  "prefix": "lib",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nx/vite:test"
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    }
  },
  "implicitDependencies": ["vite-plugin-angular"]
}
