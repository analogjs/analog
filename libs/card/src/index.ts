export * from './lib/card/card.component';
