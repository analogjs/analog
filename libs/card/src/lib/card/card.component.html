{{ title }}
<button mat-button color="primary" (click)="render = true">Render card</button>
<mat-card *ngIf="render"> Some card </mat-card>
