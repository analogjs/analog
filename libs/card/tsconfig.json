{
  "compilerOptions": {
    "forceConsistentCasingInFileNames": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "skipLibCheck": true
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    }
  ],
  "extends": "../../tsconfig.base.json",
  "angularCompilerOptions": {
    "enableI18nLegacyMessageIdFormat": false,
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
}
