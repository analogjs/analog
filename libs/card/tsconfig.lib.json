{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "declaration": true,
    "declarationMap": true,
    "inlineSources": true,
    "types": []
  },
  "exclude": ["src/**/*.spec.ts", "vite.config.ts"],
  "include": ["src/**/*.ts"]
}
