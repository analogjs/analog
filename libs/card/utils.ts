export type AngularMemoryOutputFiles = Map<
  string,
  Uint8Array
  // { contents: Uint8Array; hash: string; servable: boolean }
>;
