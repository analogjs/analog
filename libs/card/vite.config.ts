/// <reference types="vitest" />

import { defineConfig } from 'vite';
import angular from '@analogjs/vite-plugin-angular';

// https://vitejs.dev/config/
export default defineConfig(({ mode }) => {
  return {
    plugins: [angular()],
    test: {
      globals: true,
      environment: 'jsdom',
      setupFiles: ['src/test-setup.ts'],
      include: ['**/*.spec.ts'],
      cache: {
        dir: `../../node_modules/.vitest`,
      },
    },
    define: {
      'import.meta.vitest': mode !== 'production',
    },
  };
});
