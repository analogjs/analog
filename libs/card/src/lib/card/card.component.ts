import { Component } from '@angular/core';
import { CommonModule, NgIf } from '@angular/common';
import { MatCard, MatCardModule } from '@angular/material/card';

@Component({
  selector: 'lib-card',
  standalone: true,
  imports: [NgIf, MatCard],
  templateUrl: './card.component.html',
  styleUrls: ['./card.component.css'],
})
export class CardComponent {
  title = 'card-works';
  render = false;
}
