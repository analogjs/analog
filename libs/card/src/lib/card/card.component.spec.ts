import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HarnessLoader } from '@angular/cdk/testing';
import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
import { MatButtonHarness } from '@angular/material/button/testing';
import { MatCardHarness } from '@angular/material/card/testing';

import { CardComponent } from './card.component';

describe('CardComponent', () => {
  let loader: HarnessLoader;
  let fixture: ComponentFixture<CardComponent>;
  let component: CardComponent;

  beforeEach(() =>
    TestBed.configureTestingModule({
      imports: [CardComponent],
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(CardComponent);
    component = fixture.componentInstance;
    loader = TestbedHarnessEnvironment.loader(fixture);
    fixture.detectChanges();
  });

  it('should show a card once we click on the button', async () => {
    const button = await loader.getHarness(MatButtonHarness);

    const card = await loader.getHarnessOrNull(MatCardHarness);
    expect(card).toBeNull();

    await button.click();

    const cardAfterClick = await loader.getHarnessOrNull(MatCardHarness);
    expect(cardAfterClick).not.toBeNull();
  });

  it('should create the app', () => {
    expect(component).toBeTruthy();
  });

  it(`should have as title 'vitetest'`, () => {
    expect(component.title).toEqual('card-works');
  });

  it('should create the app', () => {
    expect(fixture).toMatchSnapshot();
  });
});
