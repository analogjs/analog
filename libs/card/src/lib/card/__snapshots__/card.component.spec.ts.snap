// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardComponent > should create the app 1`] = `
<lib-card>
  card-works

  <button
    _ngcontent-a-c1042141038=""
    color="primary"
    mat-button=""
  >
    Render card
  </button>
  <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
</lib-card>
`;
